plugins/c8y_configuration_plugin/src/config_manager.rs

```diff
@@ -138,17 +139,21 @@ impl ConfigManager {
                         return Ok(())
                     }
                 }
-                Some((child_id, config_type)) = self.config_upload_manager.child_ops_pending_timeout().next() => {
+                Some(((child_id, config_type), op_state)) = self.config_upload_manager.operation_timer.next_timed_out_entry() => {
+                    info!("Config snapshot request for config type: {config_type} on child device: {child_id} timed-out");
                     self.fail_pending_config_operation_in_c8y(
                         ConfigOperation::Snapshot,
                         child_id.clone(),
-                        format!("Timeout due to lack of response from child device: {} for config type: {}", child_id, config_type),
+                        op_state,
+                        format!("Timeout due to lack of response from child device: {child_id} for config type: {config_type}"),
```

Comment:
Is it too late to ask for the duplicate "Timeout due to lack of response from child device" error messages to be moved to a variable

Reply:
Will do, before the merge.

Reply:
Couldn't do it because of a Rust limitation. Even though it looks like a duplicate, the `child_id` and `config_type` variables referred to, in that error message are coming from two different `match` arms. So, I can't have this message variable created upfront. The other option is to extract the message with `{}` placeholders. But then that would have affected the readability of this code. So, keeping it as -is.